### ISSUE TITLE
memory-tests.eo:32-38: Let's make sure memory can only...

### ISSUE THREAD

**Issue by 0pdd** [objectionary/eo#2211]
The puzzle `761-525f7f4a` from #761 has to be resolved: 

https://github.com/objectionary/eo/blob/f9df2ae4e73e07ec2cb9ac7af385bf1fdedad2d6/eo-runtime/src/test/eo/org/eolang/memory-tests.eo#L32-L38

The puzzle was created by @rultor on 05-Jul-23. 

Estimate: 30 minutes,  role: DEV. 

If you have any technical questions, don't ask me, submit new tickets instead. The task will be \"done\" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

**Comment by Graur** [comment]
@yegor256 Should `tuple` saving in `memory` be prohibit?

**Comment by yegor256** [comment]
@Graur yes, only `int`, `float`, `string`, and `bool` are allowed

**Comment by Graur** [comment]
@yegor256 Should a `memory` object only work with bytes? For example:

```
memory "hello" > mem
QQ.io.stdout mem.as-string # will print "hello"
QQ.io.stdout mem # will print some bytes
```

**Comment by yegor256** [comment]
@Graur I didn't get the question. The code you suggested should work. What's wrong with it?

**Comment by Graur** [comment]
@yegor256 I would like to suggest a new implementation for the return value when storing a primitive object in `memory`. When we write `memory "hello" > mem`, the `mem` object should contain bytes instead of a string. In the current implementation of the `memory` object, it returns the string object "hello".

**Comment by yegor256** [comment]
@Graur in this case this code won't work:

```
memory 42 > x
x.write (x.plus 1)
```

Here, `x.plus` will fail with `no method .plus in bytes`. Right?

**Comment by Graur** [comment]
@yegor256 Yes, you are right

**Comment by 0pdd** [comment]
@0pdd 2 puzzles [#2434](https://github.com/objectionary/eo/issues/2434), [#2435](https://github.com/objectionary/eo/issues/2435) are still not solved.

**Comment by 0pdd** [comment]
@0pdd 2 puzzles [#2434](https://github.com/objectionary/eo/issues/2434), [#2452](https://github.com/objectionary/eo/issues/2452) are still not solved; solved: [#2435](https://github.com/objectionary/eo/issues/2435).

**Comment by 0pdd** [comment]
@0pdd the puzzle [#2434](https://github.com/objectionary/eo/issues/2434) is still not solved; solved: [#2435](https://github.com/objectionary/eo/issues/2435), [#2452](https://github.com/objectionary/eo/issues/2452).
